psutil/_common.py

```diff
@@ -169,6 +171,39 @@ def supports_ipv6():
         return False
 
 
+def parse_environ_block(data):
+    """Parse a C environ block of environment variables into a dictionary."""
+    ret = {}
+    pos = 0
+
+    def convert_string(s):
+        if PY3 and not isinstance(data, bytes):
+            return s
+        else:
+            return s.decode(sys.getfilesystemencoding())
+
+    if PY3 and not isinstance(data, bytes):
+        nul, equals = '\0', '='
+    else:
+        nul, equals = b'\0', b'='
+
+    uppercase_key = os.name == "nt"
+
+    while True:
+        next_pos = data.find(nul, pos)
+        if next_pos <= pos:
+            break
+        equal_pos = data.find(equals, pos, next_pos)
+        if equal_pos > pos:
+            key = convert_string(data[pos:equal_pos])
+            value = convert_string(data[equal_pos+1:next_pos])
+            if uppercase_key:
+                key = key.upper()
+            ret[key] = value
+        pos = next_pos + 1
+    return ret
```

Comment:
I do not think that's restricted to any particular platform.  When reading from the process memory on OSX and Windows you might get more than the environment block.  And on POSIX you can call putenv with an argument that does not contain an equals sighn.  I'm actually preparing a new version right now that will test these cases with a subprocess.